Problem5/src/problem5/Problem5.java

```diff
@@ -5,17 +5,181 @@
+                {
+                    indexes[indexLetter - 'a'] = 99999;
+                    indexLetter++;
+                }
+            }
+        }
+        catch (IOException ex) //Handle exception
+        {
+            ex.printStackTrace(System.out);
+        }
+        finally
+        {
+            try
+            {
+                if (br != null) //Close the reader
+                {
+                    br.close();
+                }
+            }
+            catch (IOException ex) //Handle exception
+            {
+                System.out.println("Error in closing the BufferedReader");
+                ex.printStackTrace(System.out);
+            }
+        }      
+    }
+    
+    
+    
+    /**
+     * Search for a word in the list
+     * 
+     * @param words List of words
+     * @param word Word to search for
+     * @param indexes The indexes of the first word to start with each letter
+     * @return The index of the word in the list
+     */
+    public static int binarySearch(String[] words, String word, int[] indexes)
+    {
+        char firstChar = word.charAt(0); //The first character in the word
+        int low, high; //The high and low indexes for binary search
+        if (firstChar == 'z') { //If the word starts with z, the starting high will be the end of the array
+            low = indexes[25];
+            high = words.length - 1;
+        } else { //Otherwise, it will be the starting index of the next letter
+            low = indexes[firstChar - 'a'];
+            high = indexes[firstChar - 'a' + 1];
+        }
+        int middle; //The middle of low and high, used to split the list
+        while (low <= high)
+        {
+            middle = (low + high)/2;
+            int compare = compareStrings(words[middle], word); //Compares the word at index middle with the word being searched
+            switch(compare)
+            {
+                case 0: //If the words are the same, return the middle index
+                    return middle;
+                case 1: //If the word at middle comes first, move low up so that you are only searching above middle
+                    low = middle+1;
+                    break;
+                case 2: //If the word at middle comes second, move high down so that you are only searching below middle
+                    high = middle-1;
+                    break;
+                default: //Exception
+                    System.out.println("Exception: String comparison invalid");
+                    return -1;
+            }
+        }
+        return -1;
+    }
+    
+    /**
+     * Compares two strings to see which comes first alphabetically
+     * 
+     * @param first The first string
+     * @param second The second string
+     * @return 0 if the strings are equal, 1 if the first string comes first, 2 if the second string comes first
+     */
+    public static int compareStrings(String first, String second)
```

Comment:
Since the words were all lower case, you could have used the compareTo method to determine if the input string was greater or less than the middle string.